bundle.proto

```diff
@@ -109,3 +109,102 @@ message BundleResult {
     Dropped dropped = 6;
   }
 }
+
+/** BundleStatus types */
+
+message FinalizedOnChain {
+  // Validator that finalized the bundle.
+  string validator_identity = 1;
+
+  // Slot it was finalized at.
+  uint64 slot = 2;
+
+  /// Index within the block.
+  uint64 bundle_index = 3;
+}
+
+message ProcessedOnChain {
+  // Validator that processed the bundle.
+  string validator_identity = 1;
+
+  // Slot it was processed at.
+  uint64 slot = 2;
+
+  /// Index within the block.
+  uint64 bundle_index = 3;
+}
+
+// Indicates the bundle won an auction and was forwarded to validators.
+message ForwardedToValidator {
+  // Slot at which bundle was forwarded.
+  uint64 slot = 1;
+
+  // Validators for which this bundle was forwarded to.
+  repeated string validators = 2;
+}
+
+// Bundle expirations are determined by their oldest blockhash.
+message BundleExpired {
+  // The blockhash that expired.
+  string blockhash = 1;
+
+  // The slot this blockhash was checked at.
+  uint64 slot_checked_at = 2;
+
+  // The expiration slot of the blockhash.
+  uint64 slot_expired_at = 3;
+}
+
+// Bundle contains transactions that have already processed on-chain.
+message ContainsProcessedTransactions {
+  repeated string processed_transactions = 1;
+}
+
+// The bundle failed one of the following validation checks.
+message ValidationFailure {
+  oneof reason {
+    SimulationFailure simulation_failure = 1;
+    BundleExpired bundle_expired = 2;
+    ContainsProcessedTransactions contains_processed_transactions = 3;
+  }
+}
+
+message Outbid {
+  // The highest bidding bundle in the auction.
+  uint64 highest_auction_bid = 1;
+
+  // Indicates whether the bundle was pruned from the mempool.
+  // e.g.
+  //  If a bundle in the batch that won the auction would invalidate this bundle then it would be pruned.
+  //  However, if this bundle was excluded from forwarding because the total forwarded bundles during the
+  //  auction window would exceed the limit, then it would be kept around in the mempool for the next auction.
+  bool pruned_from_mempool = 2;
+}
+
+message BundleStatus {
+  // Bundle's Uuid.
+  string bundle_id = 1;
+
+  // Mempool region this notification came from.
+  string mempool_region = 2;
+
+  oneof result {
+    // Bundle was forwarded to validators.
+    ForwardedToValidator forwarded_to_validator = 3;
+
+    // Bundle failed a validation check required before being inserted into the mempool.
+    ValidationFailure validation_failure = 4;
+
+    // Bundle finalized on-chain.
+    FinalizedOnChain finalized_on_chain = 5;
```

Comment:
could maybe reorder these by order of the checks. eg
1. bundle expired
2. validation fail
3. outbid
4. forwarded to validator
5. processed on chain
6. finalized on chain

Reply:
if we add a new field then order could be invalidated